### ISSUE TITLE
ProcessTest is earning its keep by failing intermittently, especially on macOS.

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2923]
I have noticed that `ProcessTest` is failing intermittently recently.  This seems to be
happening more often on macOS.  I am not sure if this started happening before
or after my recent (few weeks) `posix_spawn` changes were merged.

As frustrating as intermittent ProceesTest failures are, I suspect that the
test is validly picking up a hard-to-provoke defect in or releated to 
javalib Process.scala.

The current failures seem to be with Process trying to read from an 
InputStream, to which the/a child process is expected to have written.

The next-most recent failures tended to be on Windows and had
to do with it taking a long time (> 10 or 20 or more seconds) for
a process to exit.

Here, I think there is a race condition and/or an I/O is getting lost.
(because parent is closing wrong parts of the involved pipes?
because the child exits aborning or never exists at all?)

There may also be something happening with the whole ProcessMonitor thing.

On the whole, I think there is a real issue here. I will investigate, but wanted
others to know not to blame poor ProcessTest for sounding the alarm, albeit
in hard to understand ways.

**Comment by ekrich** [comment]
I agree. I was never that confident I could fix things there plus it would be great to get rid of the C++ as well. I thought I liked C++ along time ago when I was using it but I think it was just the Object Oriented part of it.
